### PR TITLE
DT-22: Initial CloudFront and ALB setup

### PR DESCRIPTION
**Pull request by BenRamchandani**
* Created EC2 instance running Nginx for testing purposes
* Set up ALB and CloudFront pointing at Nginx box
* Includes basic WAF rules, access logs and CloudFront -> ALB authentication with header

Only configured for the test environment.

Things left to discuss/do:
* Set up domains, both for CloudFront and the ALB, then switch the ALB over to HTTPS
* Set up custom error pages if we want those?
* If we want a maintenance mode where the ALB returns a fixed response, then set that up
* Log retention (as in deletion prevention, not expiration - that's set up) for the logs in the S3 bucket

### CODE REVIEW

**Comment by BenRamchandani** [comment]
I'll make changes in https://github.com/communitiesuk/delta-common-infrastructure/pull/9 to put Jaspersoft behind CloudFront, so probably ignore this for now

File: terraform/modules/cloudfront/cloudfront.tf [hunk 1]
```diff
@@ -1,119 +1,184 @@
...
   restrictions {
     geo_restriction {
       restriction_type = "whitelist"
-      locations        = ["US", "CA", "GB", "DE"]
+      locations        = ["GB"]
     }
   }
 
   tags = {
-    Environment = "production"
```
Comment (hugh-emerson) [inline terraform/modules/cloudfront/cloudfront.tf:113]: We already have a default tag for "environment", no?

**Review by hugh-emerson — APPROVED** [review]
Looks good.

After jaspersoft is up and running as an example origin, we can bin the nginx stuff?

**Review by hugh-emerson — CHANGES_REQUESTED** [review]
Oh actually, just noticed this is missing aws_shield_protection for the cloudfront distribution

**Comment by BenRamchandani** [comment]
> After jaspersoft is up and running as an example origin, we can bin the nginx stuff?

Yes, definitely, it's just to have something to test with

**Comment by BenRamchandani** [comment]
> Oh actually, just noticed this is missing aws_shield_protection for the cloudfront distribution. Sorry it wasn't mentioned on the ticket - adding it now.

As discussed - leaving this for DT-63

**Comment by BenRamchandani** [comment]
Test plan is failing because I'm messing around and have a bunch of us-east-1 stuff in the state. I'd like to get this merged anyway as the Jaspersoft branch builds on top of it.

**Pull request merged**